src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_Format.txt

```diff
@@ -0,0 +1,534 @@
+"Friday, Jun h. 03, 1983"
+
+>> Text(30470, "dddd, mmm ""h."" dd, yyy") 
+"Friday, Jun h. 03, 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "dddd, mmm ""hh:mm"" dd, yyy") 
+"Friday, Jun hh:mm 03, 1983"
+
+>> Text(30470, "dddd, mmm ""hh:mm"" dd, yyy") 
+"Friday, Jun hh:mm 03, 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "dddd, mmm ""hh.m"" dd, ""mm.f"" yyy") 
+"Friday, Jun hh.m 03, mm.f 1983"
+
+>> Text(30470, "dddd, mmm ""hh.m"" dd, ""mm.f"" yyy") 
+"Friday, Jun hh.m 03, mm.f 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), """hh"" dddd, mmm ""mm"" dd, ""fff"" yyy") 
+"hh Friday, Jun mm 03, fff 1983"
+
+>> Text(30470, """hh"" dddd, mmm ""mm"" dd, ""fff"" yyy") 
+"hh Friday, Jun mm 03, fff 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "dddd, mmm ""carrymefforward"" dd, yyy") 
+"Friday, Jun carrymefforward 03, 1983"
+
+>> Text(30470, "dddd, mmm ""carrymefforward"" dd, yyy") 
+"Friday, Jun carrymefforward 03, 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "dddd, mmm""carrymefforward""dd, ""fff"" yyy") 
+"Friday, Juncarrymefforward03, fff 1983"
+
+>> Text(30470, "dddd, mmm""carrymefforward""dd, ""fff"" yyy") 
+"Friday, Juncarrymefforward03, fff 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "mmm ""[some*]"" dd ""(de)+"" yyyy") 
+"Jun [some*] 03 (de)+ 1983"
+
+>> Text(30470, "mmm ""[some*]"" dd ""(de)+"" yyyy") 
+"Jun [some*] 03 (de)+ 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "mmm ([""][^""]*[""]) dd yyyy") 
+"Jun ([][^]*[""]) 03 1983"
+
+>> Text(30470, "mmm ([""][^""]*[""]) dd yyyy") 
+"Jun ([][^]*[""]) 03 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "mmm ""([""][^""]*[""]) dd yyyy") 
+"Jun ([][^]*[]) 03 1983"
+
+>> Text(30470, "mmm ""([""][^""]*[""]) dd yyyy") 
+"Jun ([][^]*[]) 03 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "mmm ""([""][^""]*[""])"" dd yyyy") 
+"Jun ([][^]*[])"" 03 1983"
+
+>> Text(30470, "mmm ""([""][^""]*[""])"" dd yyyy") 
+"Jun ([][^]*[])"" 03 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "mmm ""[de]+de"" dd yyyy") 
+"Jun [de]+de 03 1983"
+
+>> Text(30470, "mmm ""[de]+de"" dd yyyy") 
+"Jun [de]+de 03 1983"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "yyyy-mm-dd HH:mm:ss.fff") 
+"1983-06-03 02:53:09.fff"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "yyyy-mm-dd HH:mm:ss.000") 
+"1983-06-03 02:53:09.123"
+
+>> Text(30470.01953125, "yyyy-mm-dd HH:mm:ss.000") 
+"1983-06-03 00:28:07.500"
+
+>> Text(DateTime(1983, 6, 3, 2, 53, 9, 123), "YYYY-MM-DD HH:MM:SS.000") 
+"1983-06-03 02:53:09.123"
+
+>> Text(30470.01953125, "YYYY-MM-DD HH:MM:SS.000") 
+"1983-06-03 00:28:07.500"
+
+>> Text(1, 
+  "mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm")
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> Text(1, "00.00 dd-mm-yyyy")
+Errors: Error 0-27: The function 'Text' has some invalid arguments.|Warning 8-26: Incorrect format specifier for 'Text'.
+
+>> Text(1, "M#")
```

Comment:
This is a good test - let's confirm that it passes JSRunner tests in PAClient.

Reply:
Confimed